### PR TITLE
move fill_averaged_equation_of_state_outputs to utilities

### PR DESCRIPTION
**Pull request by jdannberg**
The fill_averaged_equation_of_state_outputs function is really a utilities function and should be in the material model utilities rather than in the equation of state interface. 

* [x] I have followed the [instructions for indenting my code](../blob/master/CONTRIBUTING.md#making-aspect-better).

See discussion in #4148.

### CODE REVIEW

**Comment by bobmyhill** [comment]
Cool! This is much improved :)

File: include/aspect/material_model/utilities.h [hunk 1]
```diff
@@ -317,6 +320,32 @@ namespace aspect
                             const std::vector<double> &parameter_values,
                             const CompositionalAveragingOperation &average_type);
 
+
+
+      /**
+      * This function computes averages of multicomponent thermodynamic properties
+      * that are stored in a vector of EquationOfStateOutputs.
+      * Each EquationOfStateOutput contains the thermodynamic properties for
+      * all materials at a given evaluation point.
```
Comment (gassmoeller) [inline include/aspect/material_model/utilities.h:329]: What is "Each EquationOfStateOutput"? Below we have only one "EquationOfStateOutputs" object. If you mean that one, you could write 

```suggestion
      * @p eos_outputs  contains the thermodynamic properties for
      * all materials at a given evaluation point.
```

File: source/material_model/equation_of_state/interface.cc [hunk 2]
```diff
@@ -93,11 +73,6 @@ namespace aspect
   {
 #define INSTANTIATE(dim) \
   template struct EquationOfStateOutputs<dim>; \
-  template void fill_averaged_equation_of_state_outputs<dim> (const EquationOfStateOutputs<dim> &, \
-                                                              const std::vector<double> &, \
-                                                              const std::vector<double> &, \
-                                                              const unsigned int, \
-                                                              MaterialModelOutputs<dim> &); \
```
Comment (gassmoeller) [inline source/material_model/equation_of_state/interface.cc:100]: You have removed the explicit instantiation here, but have not added it into utilities.cc that is why the unity tester failed.

**Review by gassmoeller — COMMENTED** [review]
Just one question about the documentation, and a suggestion for how to fix the tester. Ready otherwise.

**Review by gassmoeller — APPROVED** [review]

**Pull request merged**